core/spring/src/main/java/org/apache/syncope/core/spring/ImplementationManager.java

```diff
@@ -104,10 +104,19 @@
                         rule = (AccountRule) ApplicationContextProvider.getBeanFactory().
                                 getSingleton(ruleClass.getName());
                     } else {
-                        rule = (AccountRule) ApplicationContextProvider.getBeanFactory().
-                                createBean(ruleClass, AbstractBeanDefinition.AUTOWIRE_BY_TYPE, false);
-                        ApplicationContextProvider.getBeanFactory().
-                                registerSingleton(ruleClass.getName(), rule);
+                        // to avoid concurrency issues
+                        synchronized (ImplementationManager.class) {
```

Comment:
Why to synchronize on a class object? You could synchronize on `LOG`, for instance.